src/day15.scala

```diff
@@ -0,0 +1,69 @@
+// using scala 3.1.0
+
+package day15
+
+import scala.util.Using
+import scala.io.Source
+import scala.collection.mutable
+
+type Coord = (Int, Int)
+class GameMap(cells: Seq[Seq[Int]]):
+  val maxRow = cells.length - 1
+  val maxCol = cells.head.length - 1
```

Comment:
When writing Scala code, I deliberately always pick `length` over `size` – this is a good practice since `length` has a chance to invoke native operations directly, without conversion, while `size` doesn't. So if you stick to `length` all the time, you don't need to think about it and get a tiny performance boost whenever it is there...

Reply:
OK. It's your code, at the end of the day. ;)